### PR TITLE
Add support for installing local rpm packages

### PR DESCRIPTION
**Pull request by psss**
Package now can be either name or local file path to rpm.
New option -D for installing all packages from a directory.
Include a simple tier 3 test to check install using podman.

### CODE REVIEW

File: tmt/steps/prepare/install.py [hunk 1]
```diff
@@ -59,10 +80,57 @@ def go(self, guest):
...
+        local_packages = []
+        directories = self.get('directory', [])
+        if not packages and not directories:
+            self.debug("No packages for installation found.", level=3)
+            return
+
+        # Detect local packages and directories
+        for package in packages:
+            if package.endswith('.rpm'):
+                local_packages.append(package)
```
Comment (pvalena) [inline tmt/steps/prepare/install.py:95]: What if it's an URI? You can't simply copy that. It should be better detected. But I can address URIs in another PR.

**Comment by pvalena** [comment]
The install related changes LGTM.

File: tmt/steps/prepare/install.py (hunk 1 above)
Comment (psss) [reply tmt/steps/prepare/install.py:110]: Yes, for now this is adding just local files support. Let's extend this in a separate pull request.

File: tmt/steps/prepare/install.py [hunk 2]
```diff
@@ -26,44 +43,110 @@ def options(cls, how=None):
...
+        user = guest.execute('whoami')[0].strip()
+        sudo = '' if user == 'root' else 'sudo '
+        self.debug('Check if dnf is available.', level=2)
+        try:
+            guest.execute('rpm -q dnf')
+            command = f"{sudo}dnf"
+            plugin = 'dnf-plugins-core'
+        except tmt.utils.RunError:
+            command = f"{sudo}yum"
+            plugin = 'yum-plugin-copr'
```
Comment (thrix) [inline tmt/steps/prepare/install.py:86]: This won't work on rhel6/centos6, afaik there is no such package. Could we add a check somewhere in provision to abort on centos6 and smilar

Comment (psss) [reply tmt/steps/prepare/install.py:86]: But you can still install packages from non-copr repositories, which works fine:

```
> tmt run -a plan --name smoke provision -h container -i centos:6 prepare -h install -p gcc
/var/tmp/tmt/run-202

/plans/smoke
    discover
        how: shell
        directory: /home/psss/git/tmt
        summary: 1 test selected
    provision
        how: container
        image: centos:6
        distro: CentOS release 6.10 (Final)
        kernel: 5.5.11-200.fc31.x86_64
        summary: 1 guest provisioned
    prepare
        how: install
        package: gcc
        summary: 1 preparation applied
    execute
        how: shell
        summary: 1 test executed
    report
        how: display
        summary: 1 error
    finish
        container: stopped
        container: removed
        summary: 0 tasks completed
```

Why should we completely abort on rhel6/centos6?

Comment (thrix) [reply tmt/steps/prepare/install.py:86]: @psss sorry we need also support centos 6 & friends :( `convert2rhel` already uses this :(

https://github.com/oamg/convert2rhel/pull/40

 for centos 6 we can install copr builds by manually adding the repo like this:

https://gitlab.com/testing-farm/cruncher/-/blob/master/modules/cruncher.py#L399

Comment (psss) [reply tmt/steps/prepare/install.py:86]: Hm, there will need to be some special handling for group coprs:

* https://copr.fedorainfracloud.org/coprs/g/oamg/convert2rhel/repo/epel-6/group_oamg-convert2rhel-epel-6.repo

The url is different, contains an extra `/g/` and `group_` keys.

Comment (psss) [reply tmt/steps/prepare/install.py:125]: Ok, added suppport and test for `epel6` as well.

File: tmt/steps/prepare/install.py [hunk 3]
```diff
@@ -26,44 +46,140 @@ def options(cls, how=None):
...
 
         # Convert to list if necessary
-        tmt.utils.listify(self.data, split=True, keys=['package', 'copr'])
+        tmt.utils.listify(
+            self.data, split=True, keys=['package', 'directory', 'copr'])
+
+    def enable_copr_epel6(self, copr, guest):
+        """ Manually enable copr repositories for epel6 """
+        # Parse the copr repo name
+        matched = re.match("^(@)?([^/]+)/([^/]+)$", copr)
```
Comment (thrix) [inline tmt/steps/prepare/install.py:74]: seems like this would be nice as a general check @psss?

Comment (psss) [reply tmt/steps/prepare/install.py:74]: Invalid repos will be nicely handled directly in copr plugin. I don't think we need to add an extra check there.

**Pull request merged**